packages/webapp/src/components/AutoRedirect.tsx

```diff
@@ -0,0 +1,26 @@
+import type { FC } from 'react';
+import { useEffect } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { useStore } from '../store';
+import storage, { LocalStorageKeys } from '../utils/local-storage';
+
+export const AutoRedirect: FC = () => {
+    const navigate = useNavigate();
+    const location = useLocation();
+    const env = useStore((state) => state.env);
+
+    useEffect(() => {
+        if (env) {
+            storage.setItem(LocalStorageKeys.LastEnvironment, env);
```

Comment:
useLocalStorage would be nice (not that this class is bad but it's one less thing to maintain)